src/main/java/org/java_lcw/jil/AWTImage.java

```diff
@@ -0,0 +1,485 @@
+    switch(type) {
+    case JPEG: {
+      JpegFile.save(filename, this);
+    }break;
+    case PNG: {
+      PngFile.save(filename, this);
+    } break;
+    case TIFF: {
+      TiffFile.save(filename, this);
+    } break;
+    default:
+      break;
+    }
+  }
+
+
+  @Override
+  public JavaImage toJavaImage() {
+    return JavaImage.fromBufferedImage(bi);
+  }
+
+  @Override
+  public AWTImage changeMode(byte MODE) {
+    BufferedImage nbi;
+    if(MODE == Image.MODE_L) {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_BYTE_GRAY);
+    } else if(MODE == Image.MODE_RGB) {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR);
+    } else {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+    }
+    nbi.getGraphics().drawImage(bi, 0, 0, null);
+    return new AWTImage(nbi, MODE);
+  }
+
+  @Override
+  public AWTImage resizeWithBorders(int bWidth, int bHeight, Color borderColor,
+      ScaleType st) {
+    return null;
+  }
+
+  @Override
+  public AWTImage resize(int newWidth, int newHeight, boolean keepAspect, ScaleType st) {
+    if(keepAspect) {
+      int[] aspect = Utils.getAspectSize(this.width, this.height, newWidth, newHeight);
+      newWidth = aspect[0];
+      newHeight = aspect[1];
+    }
+    AWTImage nimg;
+    BufferedImage resizedImage = new BufferedImage(newWidth, newHeight, bi.getType());
+    Graphics2D g = resizedImage.createGraphics();
+
+    switch(st) {
+    case LINER:{
+      g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
+      g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_OFF);
+      g.drawImage(bi, 0, 0, newWidth, newHeight, null);
+      g.dispose();
+      nimg = new AWTImage(resizedImage, bpp);
+    } break;      
+    case CUBIC:{
+      g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BICUBIC);
+      g.drawImage(bi, 0, 0, newWidth, newHeight, null);
+      g.dispose();
+      nimg = new AWTImage(resizedImage, bpp);
+    } break;
+    case CUBIC_SMOOTH:
+      nimg = (AWTImage)Utils.biCubicSmooth(this, newWidth, newHeight);
+    default:{
+      g.setRenderingHint(RenderingHints.KEY_COLOR_RENDERING, RenderingHints.VALUE_COLOR_RENDER_SPEED);
+      g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR);
+      g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_OFF);
+      g.drawImage(bi, 0, 0, newWidth, newHeight, null);
+      g.dispose();
+      nimg = new AWTImage(resizedImage, bpp);
+    }break;
+    }
+
+    return nimg;
+  }
+
+  @Override
+  public void fillImageWithColor(Color c) {
+    Graphics2D graphics = bi.createGraphics();
+    graphics.setPaint(Utils.toAWTColor(c));
+    graphics.fillRect(0, 0, width, height);
+    graphics.dispose();
```

Comment:
I dont think so, there is not really much here that can throw.  Also Graphics .dispose are only reallu an issue until the BufferedImage is GCed, then they are all disposed.